### ISSUE TITLE
AbstractClassMirror.findMember(EObject, String) fails when inner class is used in signature of member before

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-extras#704]
AbstractClassMirror.findMember(EObject, String) fails when inner class is used in signature of member before
e.g. native boolean closeScope0(Scope scope, Scope.ScopedAccessError exception);
in ScopedMemoryAccess in Java 16

(happens e.g when running `org.eclipse.xtext.xbase.tests.serializer.SerializerTest` with Java 16

**Comment by cdietrich** [comment]
possible solution

```
	protected EObject findMember(EObject container, String fragment) {
		if (container instanceof JvmDeclaredType) {
			EList<JvmMember> members = ((JvmDeclaredType) container).getMembers();
			boolean isOp = fragment.indexOf('(') >= 0;
			for (JvmMember member : members) {
				if (!isOp) {
					if (member instanceof JvmOperation) {
						continue;
					}
				}
				String name = member.getIdentifier();
				if (equal(fragment, name)) {
					return member;
				}
			}
			for (JvmMember member : members) {
				if (!isOp) {
					if (member instanceof JvmOperation) {
						continue;
					}
				}
				String name = member.getIdentifier();
				if (fragment.startsWith(name)) {
					EObject match = findMember(member, fragment);
					if (match != null)
						return match;
				}
			}
		}
		return null;
	}
```

**Comment by cdietrich** [comment]
can be reproduced with Java 8/11 and
```
public class CachingReflectionTypeProviderTest extends ReflectionTypeProviderTest {
	
	@Test
	public void testCreateMirror_01a() {
		URI uri = URI.createURI("java:/Objects/testdata.ClientFor");
		IMirror mirror = getTypeProvider().createMirror(uri);
		TypeResource resource = (TypeResource) resourceSet.getResource(uri, true);
		assertNotNull(resource);
		assertTrue(resource.isLoaded());
		assertEquals(1, resource.getContents().size());
		for (JvmMember m : ((JvmDeclaredType)resource.getContents().get(0)).getMembers()) {
			if (m instanceof JvmField) {
				((JvmField) m).getType().getSimpleName();
			}
		}
	}
```
```
package testdata;

/**
 * @author cdietrich - Initial contribution and API
 */
public class ClientFor {
	
	
	public testdata.ScopedMemoryAccess.Scope.ScopedAccessError e;

}

```
```
package testdata;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author cdietrich - Initial contribution and API
 */
public class ScopedMemoryAccess {
	
	 private static String UNSAFE = "";

	    private static native Scope registerNatives();
	
	private ScopedMemoryAccess() {}

    private static final ScopedMemoryAccess theScopedMemoryAccess = new ScopedMemoryAccess();

    public static ScopedMemoryAccess getScopedMemoryAccess() {
        return theScopedMemoryAccess;
    }

	public boolean closeScope(Scope scope) {
		return closeScope0(scope, Scope.ScopedAccessError.INSTANCE);
	}

	native boolean closeScope0(Scope scope, Scope.ScopedAccessError exception);

	public interface Scope {
		final class ScopedAccessError {
			public static ScopedAccessError INSTANCE = new ScopedAccessError();
		}
	}
	
	@Target({ElementType.METHOD, ElementType.CONSTRUCTOR})
    @Retention(RetentionPolicy.RUNTIME)
    @interface Scoped { }
	
}
```

**Comment by cdietrich** [comment]
`org.eclipse.xtext.common.types.access.reflect.CachingReflectionTypeFactory.createType(Class<?>)` and others call 
EcoreUtil2.cloneWithProxies(cachedResult);

which wont copy identifier as it is derived.
maybe

```
EcoreUtil.Copier copier = new EcoreUtil.Copier(false) {
				 @Override
				public EObject copy(EObject eObject) {
					 EObject copy = super.copy(eObject);
					 if (copy instanceof JvmMember) {
						 copyAttribute(TypesPackage.Literals.JVM_MEMBER__IDENTIFIER, eObject, copy);
					 }
					 return copy;
				}
			};
			JvmDeclaredType copy = (JvmDeclaredType) copier.copy(cachedResult);
			copier.copyReferences();
			return copy;
```

**Comment by cdietrich** [comment]
alternative would be to change something in the metamodel. am not sure if this would have side effects

**Comment by cdietrich** [comment]
https://github.com/eclipse/xtext-extras/compare/cd_jvmTypesTest

**Comment by cdietrich** [comment]
Fixed in 2.26
